### DISCUSSION TITLE
Minimal extension?

### DISCUSSION THREAD

**Discussion by hello-users9** [zed-industries/zed#21166] [Feature Requests]
Hello. What does a minimal working extension look like?

I would like to add tree-sitter queries for more flexible highlighting of my theme. I'm not going to write anything in rust, but it looks like src/lib.rs is required. 

p.s: I'd like to add tree-sitter queries to my theme and publish it as a theme, not extension.

**Comment by SomeoneToIgnore** [comment]
I think it's totally doable without any Rust code at all, consider https://github.com/JosephTLyons/zed-brainfuck/tree/8e82d496608588f0470283ca938c87d65e48d459 as an example: it seem to contain what you want exactly, tree-sitter and extension metadata.

**Reply by hello-users9** [reply]
Well, thank you. My theme with tree-sitter queries didn't want to work a few days ago. I'll come back here later and write details.

**Reply by hello-users9** [reply]
I think I've figured out what the problem is. If I want to add a query for typescript, [grammars.typescript] “commit, repo” is required?

First I tried importing grammars from the Zed repository, naively believing that there is some default grammar there 😅. 

From a user experience perspective, I wanted to just add my query without importing a grammar (because I expected the default grammar used in Zed will work)

**Reply by SomeoneToIgnore** [reply]
Alas, no clue, I am only away of one source of the docs on the topic: https://zed.dev/docs/extensions/developing-extensions

If this is not described there, maybe it's good to open a PR.
